File: src/main/java/org/example/kalkulationsprogramm/dto/AusgangsGeschaeftsDokument/AusgangsGeschaeftsDokumentResponseDto.java
package org.example.kalkulationsprogramm.dto.AusgangsGeschaeftsDokument;

import lombok.Data;
import org.example.kalkulationsprogramm.domain.AusgangsGeschaeftsDokumentTyp;

import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * Response DTO für Ausgangs-Geschäftsdokumente.
 */
@Data
public class AusgangsGeschaeftsDokumentResponseDto {
    private Long id;
    private String dokumentNummer;
    private AusgangsGeschaeftsDokumentTyp typ;
    private LocalDate datum;
    private String betreff;
    private String htmlInhalt;
    private String positionenJson;

    // Beträge
    private BigDecimal betragNetto;
    private BigDecimal betragBrutto;
    private BigDecimal mwstSatz;
    private BigDecimal mwstBetrag;
    private Integer abschlagsNummer;
    private Integer zahlungszielTage;
    private LocalDate versandDatum;

    // Status
    private boolean gebucht;
    private LocalDate gebuchtAm;
    private boolean storniert;
    private LocalDate storniertAm;
    private boolean digitalAngenommen;
    private boolean bearbeitbar;

    // Projekt
    private Long projektId;
    private String projektBauvorhaben;
    private String projektnummer;

    // Anfrage
    private Long anfrageId;

    // Kunde (für Rechnungsadresse)
    private Long kundeId;
    private String kundennummer;
    private String kundenName;
    private String rechnungsadresse;
    /** Nur gesetzt wenn der User die Adresse manuell überschrieben hat. */
    private String rechnungsadresseOverride;

    // Vorgänger
    private Long vorgaengerId;
    private String vorgaengerNummer;

    // Ersteller
    private Long erstelltVonId;
    private String erstelltVonName;

    /**
     * Direkter PDF-URL — wird derzeit nur für virtuelle Mahn-Einträge gesetzt,
     * weil die in {@code ProjektGeschaeftsdokument} liegen und nicht über den
     * Standard-Endpoint {@code /api/ausgangs-dokumente/{id}/pdf} erreichbar
     * sind. Frontend nutzt das Feld für die PDF-Vorschau.
     */
    private String pdfUrl;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/dto/AusgangsGeschaeftsDokument/AusgangsGeschaeftsDokumentResponseDto.kt
package org.example.kalkulationsprogramm.dto.AusgangsGeschaeftsDokument

import org.example.kalkulationsprogramm.domain.AusgangsGeschaeftsDokumentTyp
import java.math.BigDecimal
import java.time.LocalDate

data class AusgangsGeschaeftsDokumentResponseDto(
    var id: Long? = null,
    var dokumentNummer: String? = null,
    var typ: AusgangsGeschaeftsDokumentTyp? = null,
    var datum: LocalDate? = null,
    var betreff: String? = null,
    var htmlInhalt: String? = null,
    var positionenJson: String? = null,
    var betragNetto: BigDecimal? = null,
    var betragBrutto: BigDecimal? = null,
    var mwstSatz: BigDecimal? = null,
    var mwstBetrag: BigDecimal? = null,
    var abschlagsNummer: Int? = null,
    var zahlungszielTage: Int? = null,
    var versandDatum: LocalDate? = null,
    var isGebucht: Boolean = false,
    var gebuchtAm: LocalDate? = null,
    var isStorniert: Boolean = false,
    var storniertAm: LocalDate? = null,
    var isDigitalAngenommen: Boolean = false,
    var isBearbeitbar: Boolean = false,
    var projektId: Long? = null,
    var projektBauvorhaben: String? = null,
    var projektnummer: String? = null,
    var anfrageId: Long? = null,
    var kundeId: Long? = null,
    var kundennummer: String? = null,
    var kundenName: String? = null,
    var rechnungsadresse: String? = null,
    var rechnungsadresseOverride: String? = null,
    var vorgaengerId: Long? = null,
    var vorgaengerNummer: String? = null,
    var erstelltVonId: Long? = null,
    var erstelltVonName: String? = null,
    var pdfUrl: String? = null,
)
